### ISSUE TITLE
Configuring number of soft keys isn't reflected on-screen

### ISSUE THREAD

**Issue by GwnDaan** [Open-Agriculture/AgIsoVirtualTerminal#18]
As suggested by the title, changing the number of soft keys in the "hardware capabilities" screen doesn't change the number of keys to the right of the terminal. Quite a few applications are designed for 12 keys instead of the 6 reported by default, so it would be useful to also test it with more than the default 😄

**Comment by martonmiklos** [comment]
Hi @ad3154 , @GwnDaan 

I started to work on adding settings for the configurable softkey dimensions.
So far works good:
![kép](https://github.com/user-attachments/assets/efd0d9f9-002d-47f2-b2f8-f33f9fcc9517)

Do you see any added value in specifying number of the key count other than the number of rows/columns?

**Comment by GwnDaan** [comment]
> Do you see any added value in specifying number of the key count other than the number of rows/columns?

I'd say if we make both the number of rows and the number of columns configurable that would be optimal. But I'm not sure if there is any requirement about this in the standard

**Comment by ad3154** [comment]
I agree with Daan. I think having both is optimal. Because the number of keys is included in a CAN message, there's value I think in being able to test how implements respond to weird numbers of keys, for example

**Comment by martonmiklos** [comment]
> But I'm not sure if there is any requirement about this in the standard

The standard covers the column layout in the terms how it should be displayed. (For me it was quite surprising that the order is right to left and top to bottom.) I have not found any messages where the row/column count would be communicated.

>   there's value I think in being able to test how implements respond to weird numbers of keys, for example

In the case if someone wants to test weird number of keys still could opt for a single column and a specific number of columns (if fits the display).

I am asking it because if we remove the keycount input we can simplify the validation (for e.g. if keycount >= row*columns).
